src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_no_gapic_config.baseline

```diff
@@ -1940,7 +1940,7 @@ public class LibraryServiceClient implements BackgroundResource {
    * single resource name type in a oneof.
    * @throws com.google.api.gax.rpc.ApiException if the remote call fails
    */
-  public final BookFromAnywhere getBookFromAnywhere(BookOneOfName name, BookName altBookName) {
+  public final BookFromAnywhere getBookFromAnywhere(BookName name, BookName altBookName) {
```

Comment:
Do we want this to become BookName? Shouldn't it be BookNameOneof?

Reply:
ok sooo previously, this was BookOne**O**f in the protofile, and now it is BookOne**o**f, which is equivalent to `book_oneof` in the GAPIC config. In Java, we take off the `_oneof` suffix from the resource name, so none of the resource name classes have "oneof"  in their names. 

Therefore the name of the resource set class is also `BookName`, which I know conflicts with the single resource entity `BookName`.

I was thinking of putting a rule in the config validator that it should be a warning if a proto package has a ResourceSet called FooOneof if there already exists a Resource called Foo. This lets gapic-generator maintain existing behavior for Java generation. Because these resource entity/set names can be overriden by the GAPIC config, the config validator would treat this as a warning instead of an error.

Thoughts?

Reply:
Hmm, I'm not sure I understand. What is the name of the oneof resource name classes that we generate currently in Java? I thought they were mostly named things like TopicOneof. How would we configure a name of TopicOneof in the annotations?

Reply:
There are no classes named `*Oneof` in google-cloud-java; there is `TopicNames` for the collection_oneof and `TopicName` for the single resource entity. The GAPIC config will list the collection_oneof names as `topic_oneof`, but we take off that "oneof" suffix ([code](https://github.com/googleapis/gapic-generator/blob/bc13b57a36471bd2af2be4739c60442d8c300fb4/src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java#L889); the comments make this sound like it was intended to have the collection oneof share the same name with a single resource name entity).

Reply:
But then a few lines below, we return the stripped name but with "name_oneof" appended: https://github.com/googleapis/gapic-generator/blob/bc13b57a36471bd2af2be4739c60442d8c300fb4/src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java#L892

That said you are completely right, we don't generate any `*Oneof` classes (we used to! my knowledge is clearly out of date). I'll take another look with this in mind. Thanks! :-)